integration/inspektor-gadget/trace_dns_test.go

```diff
@@ -121,20 +147,36 @@ func TestTraceDns(t *testing.T) {
 				e.Runtime.ContainerID = ""
 			}
 
-			ExpectEntriesToMatch(t, output, normalize, expectedEntries...)
+			compFn := func(a, b any) bool {
+				x := a.(*tracednsTypes.Event)
+				y := b.(*tracednsTypes.Event)
+				return x.Comm == y.Comm &&
+					x.Qr == y.Qr &&
+					x.Nameserver == y.Nameserver &&
+					x.PktType == y.PktType &&
+					x.DNSName == y.DNSName &&
+					x.QType == y.QType &&
+					x.Rcode == y.Rcode &&
+					x.Latency == y.Latency &&
+					x.NumAnswers == y.NumAnswers &&
+					reflect.DeepEqual(x.Addresses, y.Addresses) &&
+					x.Uid == y.Uid &&
+					x.Gid == y.Gid &&
+					x.Protocol == y.Protocol &&
+					((x.Qr == tracednsTypes.DNSPktTypeQuery &&
+						x.DstPort == y.DstPort && x.SrcIP == y.SrcIP) ||
+						(x.Qr == tracednsTypes.DNSPktTypeResponse &&
+							x.SrcPort == y.SrcPort &&
+							x.DstIP == y.DstIP))
+			}
+			ExpectedEntriesToMatchCustom(t, output, normalize, compFn, expectedEntries...)
 		},
 	}
 
-	nslookupCmds := []string{
-		fmt.Sprintf("setuidgid 1000:1111 nslookup -type=a fake.test.com. %s", dnsServer),
-		fmt.Sprintf("setuidgid 1000:1111 nslookup -type=aaaa fake.test.com. %s", dnsServer),
-	}
-
-	commands := []*Command{
+	// Start the trace gadget and verify the output.
+	commands = []*Command{
 		CreateTestNamespaceCommand(ns),
```

Comment:
At the beginning of the test, we should create the namespace and then add a cleanup function to delete the ns:

```go
	RunTestSteps([]*Command{CreateTestNamespaceCommand(ns)}, t, WithCbBeforeCleanup(PrintLogsFn(ns)))

	t.Cleanup(func() {
		RunTestSteps([]*Command{DeleteTestNamespaceCommand(ns)}, t, WithCbBeforeCleanup(PrintLogsFn(ns)))
	})
```

Next, we can create the `dnstester`, and then the `busybox`. Otherwise, if the creation of the pods fails for any reason, we won't delete the namespace.

Reply:
Done!